### ISSUE TITLE
[i18n] Language fragment identifier for conditional processing

### ISSUE THREAD

**Issue by Crissov** [w3c/svgwg#201]
SVG already supports [conditional processing](https://svgwg.org/svg2-draft/struct.html#ConditionalProcessing) based upon the environment of the renderer using `<switch>` and attributes `requiredExtensions` or `systemLanguage` in child elements. As far as I can tell, there’s no way defined for users to explicitly request a certain language when they’re embedding SVG images into their content. A straightforward way to achieve this would be fragment identifiers in URLs, which other file formats already use (e.g. `bar.pdf#page=5`):

    <img src="foo.svg#lang=en" alt="Foo in English">
    
    :root:lang(en) {background: url(foo.svg#lang=en);}

It’s obvious that BCP47 codes would be used as values. I’m not sure multiple (comma-separated?) values needed to be supported. I don’t care whether the fragment ID key actually read `lang` (as in XML and HTML), `language`, `preferredLanguage` or something else. I neither find it important whether this was defined in the core SVG spec itself or in an auxiliary standard. The use case for explicitly activated extensions seems much less important to me, but could also be defined for completeness.

Multilingual sites like Wikipedia (drawing much of its image data from central Wikicommons) could then automatically add the identifier (usually already found in the third-level domain name) or let their authors set it explicitly:

    [[File:Foo.svg#lang=en|Foo in English]]
    [[File:Foo.svg|language=English|Foo in English]]

Today, there are lots of monolingual image resources like `Foo_en.svg` instead, which can pose a maintainance concern.

**Comment by AmeliaBR** [comment]
Interesting idea.

How would you see this interacting with the existing `systemLanguage` option? Would the target fragment override anything based on user preferences?

I think this is also very closely related to a broader discussion about SVG parameters, allowing options to be set when a file is embedded.  The use of target fragments has also been discussed there, as has URL query syntax.  The problem with target fragments is that they can't be used in interactive SVG which may have internal navigation that changes the target.  The problem with URL query syntax is that it is assumed by the browser to control server-side options and therefore invalidates caching of the document.

**Comment by Crissov** [comment]
Yes, at least by default, I would prefer to respect the explicit language set by the embedding user over the implicit language set by or for the viewing user. The author is responsible for supplying elements with appropriate `systemLanguage` attributes nested in `<switch>`s.

Thank you for pointing me to [SVG Parameters](http://www.w3.org/TR/SVGParamPrimer/). I was pretty sure I wasn’t the first one with that idea, but didn’t find it before. The WD at /TR is from 2009 and there’s a [proposal by Tab](https://tabatkins.github.io/specs/svg-params/) from 2015 which focusses on passing [CSS Variables](http://www.w3.org/TR/css-variables-1/) by `foo.svg#param(key value)`. Is there any other public draft issued by the WG?

**Comment by Crissov** [comment]
There is an alternative or possibly additional approach that could cover many of the use cases I had in mind, i.e. Wikipedia: introduce an `environmentLanguage` or just `language` parameter that doesn't reflect the *browser* language but the *document* language, i.e. the state of the `:lang()` selector in the document embedding the SVG picture.
